### DISCUSSION TITLE
IsKeyPressed for some keys not working on RPi (PLATFORM DRM)

### DISCUSSION THREAD

**Discussion by larry104** [raysan5/raylib#3565] [Q&A]
I notices that the block of keys (insert, delete home, end, pup, pdown) are not working. IsKeyPressed is not getting any event for these keys (I tried different keyboards and started the application from ssh terminal vs native terminal no difference). All other keys are working fine. Does anyone have an idea what is going wrong ?

**Comment by ghost** [comment]
@larry104 Could be this https://github.com/raysan5/raylib/blob/master/src/platforms/rcore_drm.c#L167-L173 or this https://github.com/raysan5/raylib/blob/master/src/platforms/rcore_drm.c#L146-L164 I think.

**Edit:** editing.

**Reply by larry104** [reply]
Well, it might not be that but thanks to your link to the specific source code it might be this: https://github.com/raysan5/raylib/blob/a3a73b9332d0ca0d7ec81044edce31dd13d7c916/src/platforms/rcore_drm.c#L1334-L1374 since I only see F keys and up down left right keys. The scan codes for (insert, delete home, end, pup, pdown) are not mapped and are 4 byte long in contrast to 5 bytes for F keys and 3 bytes for cursor keys.

e.g. insert key is
^[[2~    27 0033 0x1b
         91 0133 0x5b
         50 0062 0x32
        126 0176 0x7e

Was there a reason not to add these keys?

**Reply by ghost** [reply]
> Was there a reason not to add these keys?

@larry104 Probably just overlooked.

**Reply by larry104** [reply]
Can this thread be converted into a bug/feature request or do I need to file this under issues?

**Reply by ghost** [reply]
@larry104 Sure, there's a _"Create issue from discussion"_ button on the right side of the page.

**Reply by larry104** [reply]
doh ... didn't see that sorry. Thanks a lot!!!
